modules/node_modules/@ncigdc/dux/cart.js

```diff
@@ -4,57 +4,189 @@
 import React from 'react';
 import { REHYDRATE } from 'redux-persist/constants';
 import _ from 'lodash';
-import UndoIcon from 'react-icons/lib/md/undo';
 
-// Custom
+import { stringify } from 'query-string';
+import { fetchApi } from '@ncigdc/utils/ajax';
 import { notify } from '@ncigdc/dux/notification';
 import { Column } from '@ncigdc/uikit/Flex';
 import { center } from '@ncigdc/theme/mixins';
+import styled from '@ncigdc/theme/styled';
 
 /*----------------------------------------------------------------------------*/
 
 const UPDATE_CART = 'UPDATE_CART';
+const ADD_TO_CART = 'ADD_TO_CART';
+const CLEAR_CART = 'CLEAR_CART';
 
-function toggleInCart(incomingFiles) {
+const NotificationColumn = styled(Column, ({
+  textAlign: 'center',
+  wordBreak: 'break-word',
+}));
+
+const getNotificationComponent = (action, id, notification, dispatch) => ({
+  action,
+  id,
+  component:
+  <NotificationColumn>
+    <span>
+      {notification.action}
+      <strong> {notification.file} </strong>
+      {notification.fileText}{notification.prepositon} the cart. {notification.extraText}
+    </span>
+    {notification.undo &&
+    <span style={center}>
+      <strong>
+        <i
+          className="fa fa-undo"
+          style={{
+            marginRight: '0.3rem',
+          }}
+        />
+        <a
+          style={{
+            textDecoration: 'underline',
+          }}
+          onClick={() => dispatch(toggleFilesInCart(notification.undo.files))}
+        >
+          Undo
+        </a>
+      </strong>
+    </span>
+    }
+  </NotificationColumn>,
+});
+
+function toggleFilesInCart(incomingFile) {
   return (dispatch, getState) => {
-    const incomingFilesArray = incomingFiles.constructor === Array ? incomingFiles : [incomingFiles];
+    const incomingFileArray = incomingFile.constructor === Array ? incomingFile : [incomingFile];
```

Comment:
Curious about the reasoning behind the naming change  

The `addAllFilesInCart` function uses plurals (`incomingFiles`, `incomingFilesArray`)
https://github.com/NCI-GDC/portal-ui/pull/1249/files#diff-e89720563aa6f742cd9e3d9c264e4511R122